+ source ../common/common_lib.sh
++ source /root/mugen-riscv/libs/locallibs/common_lib.sh
+++ python3 --version
+++ '[' 0 -eq 0 ']'
+++ source /root/mugen-riscv/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Sat Nov 12 02:00:09 2022 - INFO  - Start environmental preparation.
+ DNF_INSTALL 'clamav clamd'
+ pkgs='clamav clamd'
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py install --pkgs 'clamav clamd' --node 1 --tempfile ''
+ tmpfile2=/tmp/tmp0hy05gv8
+ '[' -z '' ']'
+ tmpfile=/tmp/tmp0hy05gv8
+ echo 'LogSyslog yes
TCPSocket 3310
TCPAddr 127.0.0.1
User clamscan
OnAccessIncludePath /home'
+ SLEEP_WAIT 30
+ wait_time=30
+ /usr/sbin/clamd
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 30 --cmd '' --mode 1
LibClamAV Warning: **************************************************
LibClamAV Warning: ***  The virus database is older than 7 days!  ***
LibClamAV Warning: ***   Please update it as soon as possible.    ***
LibClamAV Warning: **************************************************
+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Sat Nov 12 02:03:30 2022 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start testing...'
+ message='Start testing...'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start testing...'
Sat Nov 12 02:03:32 2022 - INFO  - Start testing...
+ test_execution clamav-clamonacc.service
+ service=clamav-clamonacc.service
++ date '+%Y-%m-%d %T'
+ log_time='2022-11-12 02:03:32'
+ test_restart clamav-clamonacc.service
+ service=clamav-clamonacc.service
+ systemctl restart clamav-clamonacc.service
+ CHECK_RESULT 0 0 0 'clamav-clamonacc.service restart failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='clamav-clamonacc.service restart failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status clamav-clamonacc.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'clamav-clamonacc.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='clamav-clamonacc.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'clamav-clamonacc.service restart failed'
+ LOG_ERROR 'clamav-clamonacc.service restart failed'
+ message='clamav-clamonacc.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'clamav-clamonacc.service restart failed'
Sat Nov 12 02:03:40 2022 - ERROR - clamav-clamonacc.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 37'
+ message='../common/common_lib.sh line 37'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 37'
Sat Nov 12 02:03:41 2022 - ERROR - ../common/common_lib.sh line 37
+ return 0
+ systemctl stop clamav-clamonacc.service
+ CHECK_RESULT 0 0 0 'clamav-clamonacc.service stop failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='clamav-clamonacc.service stop failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status clamav-clamonacc.service
+ grep 'Active: inactive'
+ CHECK_RESULT 1 0 0 'clamav-clamonacc.service stop failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='clamav-clamonacc.service stop failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'clamav-clamonacc.service stop failed'
+ LOG_ERROR 'clamav-clamonacc.service stop failed'
+ message='clamav-clamonacc.service stop failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'clamav-clamonacc.service stop failed'
Sat Nov 12 02:03:49 2022 - ERROR - clamav-clamonacc.service stop failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 42'
+ message='../common/common_lib.sh line 42'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 42'
Sat Nov 12 02:03:50 2022 - ERROR - ../common/common_lib.sh line 42
+ return 0
+ systemctl start clamav-clamonacc.service
+ CHECK_RESULT 0 0 0 'clamav-clamonacc.service start failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='clamav-clamonacc.service start failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status clamav-clamonacc.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'clamav-clamonacc.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='clamav-clamonacc.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'clamav-clamonacc.service start failed'
+ LOG_ERROR 'clamav-clamonacc.service start failed'
+ message='clamav-clamonacc.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'clamav-clamonacc.service start failed'
Sat Nov 12 02:03:58 2022 - ERROR - clamav-clamonacc.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 47'
+ message='../common/common_lib.sh line 47'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 47'
Sat Nov 12 02:03:59 2022 - ERROR - ../common/common_lib.sh line 47
+ return 0
+ test_enabled clamav-clamonacc.service
+ service=clamav-clamonacc.service
++ systemctl is-enabled clamav-clamonacc.service
+ state=disabled
+ '[' disabled == enabled ']'
+ '[' disabled == disabled ']'
++ systemctl enable clamav-clamonacc.service
++ awk '{print $3}'
++ grep 'Created symlink'
+ symlink_file=/etc/systemd/system/multi-user.target.wants/clamav-clamonacc.service
+ find /etc/systemd/system/multi-user.target.wants/clamav-clamonacc.service
/etc/systemd/system/multi-user.target.wants/clamav-clamonacc.service
+ CHECK_RESULT 0 0 0 'clamav-clamonacc.service enable failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='clamav-clamonacc.service enable failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ systemctl disable clamav-clamonacc.service
Removed /etc/systemd/system/multi-user.target.wants/clamav-clamonacc.service.
+ find /etc/systemd/system/multi-user.target.wants/clamav-clamonacc.service
find: ‘/etc/systemd/system/multi-user.target.wants/clamav-clamonacc.service’: No such file or directory
+ CHECK_RESULT 1 0 1 'clamav-clamonacc.service disable failed'
+ actual_result=1
+ expect_result=0
+ mode=1
+ error_log='clamav-clamonacc.service disable failed'
+ '[' -z 1 ']'
+ '[' 1 -eq 0 ']'
+ test 1x == 0x
+ return 0
+ journalctl --since '2022-11-12 02:03:32' -u clamav-clamonacc.service
+ grep -i 'fail\|error'
+ grep -v -i 'DEBUG\|INFO\|WARNING'
Nov 12 02:03:32 openEuler-riscv64 clamonacc[2185]: ERROR: Clamonacc: fanotify_init failed: Function not implemented
Nov 12 02:03:32 openEuler-riscv64 systemd[1]: clamav-clamonacc.service: Failed with result 'exit-code'.
Nov 12 02:03:51 openEuler-riscv64 clamonacc[2201]: ERROR: Clamonacc: fanotify_init failed: Function not implemented
Nov 12 02:03:51 openEuler-riscv64 systemd[1]: clamav-clamonacc.service: Failed with result 'exit-code'.
+ CHECK_RESULT 0 0 1 'There is an error message for the log of clamav-clamonacc.service'
+ actual_result=0
+ expect_result=0
+ mode=1
+ error_log='There is an error message for the log of clamav-clamonacc.service'
+ '[' -z 0 ']'
+ '[' 1 -eq 0 ']'
+ test 0x == 0x
+ test -n 'There is an error message for the log of clamav-clamonacc.service'
+ LOG_ERROR 'There is an error message for the log of clamav-clamonacc.service'
+ message='There is an error message for the log of clamav-clamonacc.service'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'There is an error message for the log of clamav-clamonacc.service'
Sat Nov 12 02:04:16 2022 - ERROR - There is an error message for the log of clamav-clamonacc.service
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 28'
+ message='../common/common_lib.sh line 28'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 28'
Sat Nov 12 02:04:18 2022 - ERROR - ../common/common_lib.sh line 28
+ return 0
+ test_reload clamav-clamonacc.service
+ service=clamav-clamonacc.service
+ systemctl start clamav-clamonacc.service
+ systemctl reload clamav-clamonacc.service
+ grep 'Job type reload is not applicable'
Failed to reload clamav-clamonacc.service: Job type reload is not applicable for unit clamav-clamonacc.service.
+ CHECK_RESULT 0 0 0 'Job type reload is not applicable for unit clamav-clamonacc.service'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Job type reload is not applicable for unit clamav-clamonacc.service'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ systemctl status clamav-clamonacc.service
+ grep 'Active: active'
+ systemctl status clamav-clamonacc.service
+ grep 'inactive (dead)'
+ return 1
+ LOG_INFO 'Finish test!'
+ message='Finish test!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish test!'
Sat Nov 12 02:04:20 2022 - INFO  - Finish test!
+ CASE_RESULT 0
+ case_re=0
+ test -z 4
+ test 4 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Sat Nov 12 02:04:21 2022 - ERROR - failed to execute the case.
+ exit 4
+ post_test
+ LOG_INFO 'start environment cleanup.'
+ message='start environment cleanup.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'start environment cleanup.'
Sat Nov 12 02:04:22 2022 - INFO  - start environment cleanup.
++ pgrep -f clamd
+ kill -9 2173 2175
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z /tmp/tmp0hy05gv8 ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile /tmp/tmp0hy05gv8
/root/mugen-riscv/libs/locallibs/common_lib_python.sh: line 97:  2173 Killed                  /usr/sbin/clamd
+ '[' 0 -ne 0 ']'
+ LOG_INFO 'Finish environment cleanup!'
+ message='Finish environment cleanup!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish environment cleanup!'
Sat Nov 12 02:05:00 2022 - INFO  - Finish environment cleanup!
